[fieash] iP

Fieash opened the pull request:
GoonBot
> "Object-oriented programming is an exceptionally bad idea which could only have originated in California." – Edsger W. Dijkstra ([source](https://wiki.c2.com/?EwDijkstraQuotes))

GoonBot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~SLOW~~ **SUPER FAST** to use
- All you need to do is,

download it from [here](https://nus-cs2103-ay2425s2.github.io/website/admin/ip-w4.html).

1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
4. And it is FREE!

Features:
- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)


```java
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Fieash committed:
level 0

Fieash committed:
level 1 echo

Fieash committed:
level 2 list adding

Fieash committed:
level3 mark as done

Fieash committed:
level 3, added missing unmark feature

Fieash committed:
refactor to use Task class

Fieash committed:
level 4, created deadline event todo and functioned out reusable code

Fieash committed:
modify runtest to accomodate for new test cases

Fieash committed:
catch basic input length errors

Fieash committed:
add Delete feature

Fieash committed:
add generic GoonException, no implementation in code

Fieash committed:
add new GoonException

Fieash committed:
add auto save to file feature

Fieash committed:
Merge branch 'master' into branch-Level-7

Fieash committed:
Revert "Merge branch 'master' into branch-Level-7"

This reverts commit e57634b913e705d67246c0b40036bac7bfccc528, reversing
changes made to c037fc298c3e5b12d70a79ef923f2a382ce0bc60.

Fieash committed:
Merge commit 'c037fc298c3e5b12d70a79ef923f2a382ce0bc60'

leedeen01 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + "(by:" + by + ")";
```
Perhaps a adding javadocs will help improve code clarity and help users understand the implementation?

leedeen01 commented on src/main/java/Goon.java:
```diff
@@ -0,0 +1,217 @@
+import java.io.FileNotFoundException;
+
+public class Goon {
+    public static void formattedPrint(String input) {
```
Any reason why you wrote all the UI this way instead of making it a separate class?

leedeen01 reviewed (commented):
LGTM
Javadocs requested

hsinyilow commented on src/main/java/Goon.java:
```diff
@@ -0,0 +1,217 @@
+                } else if (tasks[0].contains("D")) { //deadline case
+                    String desc = tasks[2].split("/by")[0].substring(1);
+                    String by = tasks[2].split("/by")[1];
+                    Deadline newDead = new Deadline(desc, by);
```
Any reason why you name the task `newDead` instead of `newDeadline`?

hsinyilow commented on src/main/java/Goon.java:
```diff
@@ -0,0 +1,217 @@
+        } catch (FileNotFoundException e) {
+            System.out.println("Error: Could not find the tasks.txt file.");
+            try {
+                File nf = new File("data/tasks.txt");
```
Should this variable be abbreviated?

hsinyilow commented on src/main/java/Goon.java:
```diff
@@ -0,0 +1,217 @@
+        }
+    }
+
+    public static void deleteTask(int taskIndex, ArrayList<Task> listInputs) {
+        printDivider("\tNoted. I've removed this task:");
+        Task taskToDelete = listInputs.get(taskIndex - 1);
+        listInputs.remove(taskIndex - 1);
+        System.out.println("\t" + taskToDelete.toString());
+        System.out.println("\tNow you have " + countTasks(listInputs) + " tasks in the list.");
+        printDivider("");
+    }
+
+    public static boolean markCheck(int actual, int minimum, String taskType) {
+        if (actual < minimum) {
+            formattedPrint("Gooner, " + taskType + "ing of a task needs a number, have you attended primary school? idiot.");
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean descriptionCheck(int actual, int minimum, String taskType) {
```
Good method naming throughout the class.

hsinyilow commented on src/main/java/Goon.java:
```diff
@@ -0,0 +1,217 @@
+                    if(!markCheck(input.length(), 5, "mark")) {
+                        continue;
+                    }
+                    int mark = input.charAt(5) - '0';
+                    printDivider("\tNice! I've marked this task as done:");
+                    Task taskToMark = listInputs.get(mark - 1);
+                    listInputs.set(mark - 1, taskToMark.markAsDone());
+                    System.out.println("\t" + taskToMark.toString());
+                    printDivider("");
+
+                } else if (input.startsWith("unmark")) { //unmarking tasks
+                    if(!markCheck(input.length(), 6, "mark")) {
+                        continue;
+                    }
+                    int unmark = input.charAt(7) - '0';
+                    printDivider("\tOK, I've marked this task as not done yet:");
+                    Task taskToUnmark = listInputs.get(unmark - 1);
+                    listInputs.set(unmark - 1, taskToUnmark.unmarkAsDone());
+                    System.out.println("\t" + taskToUnmark.toString());
+                    printDivider("");
```
I like the naming in this section, it is very clear what each line of code is doing.

Fieash committed:
add LocalDate format to deadline class

Fieash committed:
update tasks.txt to new format

Fieash committed:
Merge branch 'branch-Level-8'

Fieash committed:
fix wrong format of deadline being written to file

Fieash committed:
convert to OOP format with new classes Ui, Storage, Parser, TaskList

Fieash committed:
Merge branch 'add-gradle-support'

Fieash committed:
clean up commented code from before OOPing the project

Fieash committed:
update gradle build file

Fieash committed:
add JavaDoc documentation to all classes

Fieash committed:
fix coding standard violations

Fieash committed:
add find function for level 9

Fieash committed:
Merge branch 'branch-A-JavaDoc'

Fieash committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts (resolved):
#	src/main/java/goon/Ui.java
#	src/main/java/goon/tasks/TaskList.java

Fieash committed:
Merge branch 'branch-Level-9'

# Conflicts (resolved):
#	gradle/wrapper/gradle-wrapper.properties
#	src/main/java/goon/tasks/Task.java
#	src/main/java/goon/tasks/TaskList.java

Fieash committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/goon/GoonException.java

Fieash committed:
Fix missing findTask function

> was likely missing due to a misclick in a previous merge

Fieash committed:
add simple UI for Goon following tutorial

Fieash committed:
# update to styled fxml
- styled javafx
- new avatars

Fieash committed:
link the UI to the chatbot

Fieash committed:
Merge branch 'branch-Level-10'

Fieash committed:
add JUnit tests

Fieash committed:
add new Goon image

Fieash committed:
Ui class: assert the length of user's inputs for making and unmarking
Task class: remove the unusused getDescription() method, assert description to not be null

ensures that when there are issues with userinput for length, it is caught.

an empty description could lead to a buggy Task when parsing

Let's,
* assert the length of the user's inputs, to ensure commands are properly executed
* assert that the description is filled

Fieash committed:
Refactor code to use the Parser class

Was previously dumping all logic of input and output in the UI class.

Extracted the parsing of inputs into the Parser class, formatting and passing the output back to the UI class, letting UI class do the exception handling and calling only.

This is the ideal way OOP should be done in this case, however we may still further SLAP the parser class.

* also renamed unmarkAsDone() to unmark() because it didn't make sense

Fieash committed:
Merge pull request #2 from Fieash/branch-A-Assertions

Ui class: assert the length of user's inputs for making and unmarking

Fieash committed:
Merge branch 'master' into branch-A-CodeQuality

Fieash committed:
Merge pull request #3 from Fieash/branch-A-CodeQuality

Refactor code to use the Parser class

Fieash committed:
extend based on the BCD-Extension selected, D-Contacts

* added support for storing contacts into the taskList

Fieash committed:
Merge branch 'branch-BCD-Extension'

Fieash committed:
add Ui Screenshot

Fieash committed:
update README.md to include user guide

* guide is also hosted at https://fieash.github.io/ip

Fieash committed:
reverse previous update on wrong README

*updated the correct user guide README under docs/README

Fieash committed:
fix logical erros

* commands were not actually interacting with the file store
* updated the interaction so that IO operations are being carried out every time

Fieash committed:
# reformat code to be compliant to coding style

code is not compliant with the coding style provided

we need to make it easier to read

we will use checkstyle to check the code agianst standard violations

this is done so that we can write maintainable code that others can pick up in the future and work on

Fieash committed:
Update README.md on docs with summary of commands

hansherry committed:
Revert "Update README.md on docs with summary of commands"

This reverts commit 6bc4e7daae5a560542701a99e78f6c5256dc2157.

Fieash committed:
Update README.md to include summary of commands

hansherry committed:
Add command summary to README.md

hansherry committed:
Update README.md
